app/models/user.rb

```diff
@@ -304,16 +304,16 @@ def name(formatter = nil)
     end
   end
 
-  # Return user's authentication provider for display
   def authentication_provider
-    return if identity_url.blank?
+    return nil if identity_url.blank?
 
-    identity_url.split(":", 2).first
+    slug = identity_url.split(":", 2).first
+    AuthProvider.find_by(slug:)
   end
 
   # Return user's authentication provider for display
   def human_authentication_provider
-    authentication_provider&.titleize
+    authentication_provider&.slug&.titleize
```

Comment:
While this change is the semantically equal refactoring of the previous code, I am wondering whether we should now use this instead:

```suggestion
    authentication_provider&.display_name
```

This seems more logical to me, now that we have loaded the correct provider. Though it would change what's displayed. E.g. if you have `Keycloak` as an OIDC provider, it would have previously been shown as `Oidc Keycloak`, while it's now shown as `Keycloak` :shrug:

Reply:
Looking at where/how this is used, I am now using the `display_name`.